### PR TITLE
add trace!() instrumentation

### PR DESCRIPTION
**Pull request by swaits**
- Move some debug/info into trace.
- Add trace in a few spots related to `use` and references

### CODE REVIEW

File: atelier-core/src/builder/mod.rs [hunk 1]
```diff
@@ -242,6 +245,18 @@ impl ModelBuilder {
             {
                 self.prelude_namespace.make_shape(shape_name)
             } else {
+                // we couldn't find this shape name
```
Comment (johnstonskj) [inline atelier-core/src/builder/mod.rs:248]: Can you wrap this in a debug_assertion check please

`if cfg!(debug_assertions) {`

File: atelier-core/src/builder/mod.rs [hunk 2]
```diff
@@ -242,6 +245,18 @@ impl ModelBuilder {
...
+                let mut known_shape_names = prelude::prelude_model_shape_ids(&self.smithy_version)
+                    .iter()
+                    .map(|s| format!("{}", s))
+                    .collect::<Vec<String>>();
+                known_shape_names.sort();
+                debug!(
```
Comment (johnstonskj) [inline atelier-core/src/builder/mod.rs:254]: wouldn't this be error? OR have a smaller error message and the longer debug to follow.

File: atelier-assembler/src/lib.rs [hunk 3]
```diff
@@ -435,7 +435,7 @@ impl ModelAssembler {
     }
 
     fn read_model(&self, path: &Path) -> Result<Model> {
-        info!("ModelAssembler::read_model({:?})", path);
+        trace!("ModelAssembler::read_model({:?})", path);
```
Comment (johnstonskj) [inline atelier-assembler/src/lib.rs:438]: I can see moving some debug messages to trace, but info?

**Review by johnstonskj — CHANGES_REQUESTED** [review]
Comments in the review

File: atelier-assembler/src/lib.rs (hunk 3 above)
Comment (swaits) [reply atelier-assembler/src/lib.rs:438]: I treat `trace` as things that trace the execution path, which conventionally for me falls into the "we entered this function" or "we entered this branch".

But that's a me'ism. Glad to revert all these. It was just a way for me to reduce (and then focus on the right) the noise while debugging, so "RUST_LOG=info" got quite a bit quieter.

Comment (johnstonskj) [reply atelier-assembler/src/lib.rs:438]: The only issue I have is that trace is two levels lower, I like info for entry-point functions, I want to see the shape of the execution and have debug and trace for the details

**Comment by swaits** [comment]
This PR is no longer relevant. Closing.

**Pull request closed without merging**